lib/validate-selectors.js

```diff
@@ -3,6 +3,33 @@ var shouldIgnoreRule = require('./should-ignore-rule');
 var shouldIgnoreSelector = require('./should-ignore-selector');
 var toInterpoloatedRegexp = require('./to-interpolated-regexp');
 
+function isTopLevel(node) {
+  var type = node.parent.type;
+  return type === 'atrule' || type === 'root';
+}
+
+function isNestedRule(node) {
+  return node.parent.type === 'rule';
+}
+
+function onlyHasChildRules(node) {
+  return node.nodes.length && node.every(function(child) {
+    return child.type === 'rule';
+  });
+}
+
+function unWrapSelectors(parent, child) {
```

Comment:
I should have suggested this: https://github.com/davidtheclark/postcss-resolve-nested-selector. Do you that module this will work for this case? We use it in stylelint for rules about nested selectors.